src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,429 @@
+`mainAxisAlignment` is a property that determines how `Row` and `Column` can position their children *horizontally* in that extra space. `mainAxisAlignment` has six `MainAxisAlignment` enums: 
+
+`MainAxisAlignment.start`
+: Positions children near the beginning of the main axis. (Left for `Row`, top for `Column`)
+
+`MainAxisAlignment.end`
+: Positions children near the end of the main axis. (Right for `Row`, bottom for `Column`)
+
+`MainAxisAlignment.center`
+: Positions children at the middle of the main axis.
+
+`MainAxisAlignment.spaceBetween`
+: Divides the extra space evenly between children.
+
+`MainAxisAlignment.spaceEvenly`
+: Divides the extra space evenly between children and creates an even amount of space before the first child and after the last child.
+
+`MainAxisAlignment.spaceAround`
+: Similar to `MainAxisAlignment.spaceEvenly` but reduces half of the space before the first child and after the last child. 
+
+#### Code example: mainAxisAlignment property
+{:.no_toc}
+{{site.alert.secondary}}
+  Try Dart.
+
+  Notice how `mainAxisAlignment` is set to `MainAxisAlignment.start`.
+
+  **1.** Click the **Run** button.
+
+  **2.** Change the `mainAxisAlignment` property to `MainAxisAlignment.end`.
+
+  **3.** Click the **Run** button. 
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/cb8abed13f90a6a0c7a0ada6f15a09c9
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=cb8abed13f90a6a0c7a0ada6f15a09c9&amp;fw=true&amp;split=60" width="100%" height="400px"></iframe>
+{{site.alert.tip}}
+  Before moving to the next section, try changing `MainAxisAlignment.start` to one of the other enums. 
+{{site.alert.end}}
+
+## crossAxisAlignment property
+
+Similar to the `mainAxisAlignment` property, 
+`crossAxisAlignment` is a property that determines how `Row` and `Column` can position their children in extra space. 
+However, 
+the `crossAxisAlignment` property determines how `Row` and `Column` can position their children *vertically* in that extra space. 
+Note that most of the `crossAxisAlignment` property's enums only function inside of `Row`. 
+The `crossAxisAlignment` property has five `CrossAxisAlignment` enums:
+
+`CrossAxisAlignment.start`
+: Positions children near the top of the cross axis. (`Row`)
+
+`CrossAxisAlignment.end`
+: Positions children near the bottom of the cross axis. (`Row`)
+
+`CrossAxisAlignment.center`
+: Positions children at the middle of the cross axis. (`Row`)
+
+`CrossAxisAlignment.stretch`
+: Stretches children across the cross axis. (Top to bottom for `Row`, left to right for `Column`)
+
+`CrossAxisAlignment.baseline`
+: Aligns children by their character baselines. (`Text` class)
+
+{{site.alert.tip}}
+  `CrossAxisAlignment.baseline` functions with `Text` classes and requires the `textBaseline` property set to `TextBaseline.alphabetic`. In the [Text class](#text-class) section, 
```

Comment:
functions with => works only with

and requires the => and requires that the `textBaseLine` property is set to...

In the ...  => See the [Text class](#text-class) section for an example that uses this value.

Reply:
Changed.